src/frontend/js/project-service.js

```diff
@@ -36,6 +37,17 @@ angular.module('seashell-projects', ['seashell-websocket', 'marmoset-bindings'])
       var PROJ_FILE_LIST_URL_TEMPLATE = CGI_URL + "skeleton_file_list.rkt?template=%s";
       var PROJ_WHITE_LIST_URL = CGI_URL + "project_whitelist.cgi";
       var USER_WHITE_LIST_URL = CGI_URL + "user_whitelist.cgi";
+
+      localfiles.init().then(function () { 
+        ws.register_callback("connected", function () {
```

Comment:
Note: we should probably not suppress the sync for the "forced" offline mode, as it puts a strange dependency between the websocket service and the settings service.  Additionally, we should probably handle the initialization of localfiles in websocket-service.js, so that we trigger the "connected" callback when the websocket is ready (connected, all files synced).